extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -236,12 +236,26 @@ class LifecycleTask extends BackbeatTask {
                 }
             } else {
                 // 2. by VersionId, lower number means newer
+                //    NOTE: VersionId may be null
+                const doesVersionIdExist = (versions[vIdx].VersionId !== 'null'
```

Comment:
Maybe I'm being nitpicky, but the variable name does not sound so helpful, since a version ID does exist somewhere even though there's a null version on the other side (and we can argue that null is also a version ID), why not calling it e.g. `noNullVersion`?